rankers/BM25Ranker/__init__.py

```diff
@@ -9,42 +9,52 @@
 
 class BM25Ranker(Chunk2DocRanker):
     """
-    :class:`BM25Ranker` calculates the weighted score from the matched chunks. The weights of each chunk is based on
-        the tf-idf algorithm. Each query chunk is considered as a ``term``, and the frequency of the query chunk in a
-        specific matched document is considered as the naive ``term-frequency``. All the matched results as a whole is
-        considered as the corpus, and therefore the frequency of the query chunk in all the matched docs is considered
-        as the naive ``document-frequency``. Please refer to the functions for the details of calculating ``tf`` and
-        ``idf``.
+    :class:`BM25Ranker` calculates the weighted score from the matched chunks.
+        The weights of each chunk is based on the tf-idf algorithm.
+        Each query chunk is considered as a ``term``,
+        and the frequency of the query chunk in a specific matched document
+        is considered as the naive ``term-frequency``.
+        All the matched results as a whole are considered as the corpus,
+        and therefore the frequency of the query chunk in all the matched docs
+        is considered as the naive ``document-frequency``.
+        Please refer to the functions for the details
+        of calculating ``tf`` and ``idf``.
+
+    :param k: the parameter ``k`` for the BM25 algorithm.
+    :param b: the parameter ``b`` for the BM25 algorithm.
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
     """
     required_keys = {'length', }
 
     def __init__(self, k=1.2, b=0.75, threshold=0.1, *args, **kwargs):
-        """
-
-        :param k: the parameter ``k`` for the BM25 algorithm.
-        :param b: the parameter ``b`` for the BM25 algorithm.
-        """
+        """Set constructor."""
```

Comment:
Remove this docstring completely. According to our updated guidelines, `__init__` function docstring should completely go into the class definition.

Reply:
Yes all the params of the contructor have to be on the class, but we still need a docstring specifying this for the constructor right? As in their docs: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

Reply:
According to the sphinx documentations (https://www.sphinx-doc.org/ar/stable/usage/extensions/example_google.html#example-google) you either do class and `__init__` documentation or everything in the class docstring. You don't mix it (e.g. putting parameters of `__init__` into class and still write something for the constructor itself).

Reply:
@maximilianwerk but that's for the Google style (using Sphinx, but not "sphinx style") right? I can remove it but then I get a warning from the `pydocstyle`, are we ok with that?

Reply:
OK, you convinced me - let's keep it. I start to understand why google style might be the way to go at some point in time.

Reply:
Could you approve the changes to merge? :)